test/jsn_tests.erl

```diff
@@ -123,6 +123,8 @@ new_2_test_() ->
     Object2Plist = [{<<"foo">>, [{<<"bar">>, <<"hoge">>}]}],
     Object2Eep18 = {[{<<"foo">>, {[{<<"bar">>, <<"hoge">>}]}}]},
     Object2Struct = {struct, [{<<"foo">>, {struct, [{<<"bar">>, <<"hoge">>}]}}]},
+    Object3Struct = {struct, [{<<"foo">>, [{struct, [{<<"bar">>, <<"baz1">>}]},
```

Comment:
I like this test case—an array with objects instead of "simple" terms inside the array. can you continue the pattern with `Object1...` and `Object2...` to span the supported formats, please?